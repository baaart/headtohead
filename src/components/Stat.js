import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: start;
  width: 100%;
  margin-bottom: 0;
  padding: 0;
`
const Label = styled.p`
  text-align: left;
  font-weight: 600;
  position: relative;
  min-width: 80px;
  margin-bottom: 0;

  &:after {
    content: ':';
    position: absolute;
    right: 0;
  }
`
const Value = styled.p`
  font-weight: 400;
  margin: 0;
  margin-left: 1em;
  text-align: left;
`

function Stat({...props}) {
  return (
    <Wrapper >
      <Label>{props.label}</Label>
      <Value>{props.value}</Value>
    </Wrapper>
   );
}

export default Stat;
