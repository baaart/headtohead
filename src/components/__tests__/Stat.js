import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import Stat from '../Stat';
import { fakeData } from '../__mocks__/fakeData'

describe('Stat', () => {
  it('Matches snapshot', () => {
    const wrapped = shallow(<Stat label="Height" value="183"/>);
    expect(wrapped).toMatchSnapshot();
  })
});
