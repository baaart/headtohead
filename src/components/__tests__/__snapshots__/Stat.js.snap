// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stat Matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Stat
    formatter={[Function]}
    label="Height"
    value="183"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef>
          Height
        </ForwardRef>,
        <ForwardRef>
          183
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Height",
        },
        "ref": null,
        "rendered": "Height",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
  text-align: left;
  font-weight: 600;
  position: relative;
  min-width: 80px;
  margin-bottom: 0;

  &:after {
    content: ':';
    position: absolute;
    right: 0;
  }
",
            ],
          },
          "displayName": "styled.p",
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "p",
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "183",
        },
        "ref": null,
        "rendered": "183",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": true,
            "rules": Array [
              "
  font-weight: 400;
  margin: 0;
  margin-left: 1em;
  text-align: left;
",
            ],
          },
          "displayName": "styled.p",
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "p",
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  display: flex;
  flex-direction: row;
  justify-content: start;
  width: 100%;
  margin-bottom: 0;
  padding: 0;
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef>
            Height
          </ForwardRef>,
          <ForwardRef>
            183
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Height",
          },
          "ref": null,
          "rendered": "Height",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "rules": Array [
                "
  text-align: left;
  font-weight: 600;
  position: relative;
  min-width: 80px;
  margin-bottom: 0;

  &:after {
    content: ':';
    position: absolute;
    right: 0;
  }
",
              ],
            },
            "displayName": "styled.p",
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "p",
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "183",
          },
          "ref": null,
          "rendered": "183",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-htpNat",
              "isStatic": true,
              "rules": Array [
                "
  font-weight: 400;
  margin: 0;
  margin-left: 1em;
  text-align: left;
",
              ],
            },
            "displayName": "styled.p",
            "render": [Function],
            "styledComponentId": "sc-htpNat",
            "target": "p",
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  display: flex;
  flex-direction: row;
  justify-content: start;
  width: 100%;
  margin-bottom: 0;
  padding: 0;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
